data_tests.r

```diff
@@ -0,0 +1,95 @@
+# This file contains tests to be applied to 
+# the Vegvesen stations-data *after* being transformed
+# to a data frame. 
+# 
+# All tests are packed in a function test_stations_metadata that apples
+# all the aforementioned tests
+
+#Checking if the columns names in the metadata matches column names
+#"id, name etc."
+
+test_stations_metadata_colnames <-
+  function(df) {
+    
+    expected_colnames <- c("id", "name", "latestData", "lat", "lon")
+    
+    if (all(colnames(df) == expected_colnames) == TRUE) {
+      print("PASS: Data has the correct columns")
+    } else{
+      print("FAIL: Columns do not match the correct specification")
+    }
+  }
+
+#Checking if the metadata conists of a reasonable number of rows
+
+test_stations_metadata_nrows <-
+  function(df) {
+    
+    min_expected_rows <- 5000
+    max_expected_rows <- 10000
+    
+    if (nrow(df) > min_expected_rows & nrow(df) < max_expected_rows) {
+      print("PASS: Data has a reasonable number of rows")
+    } else if (nrow(df) <= min_expected_rows) {
+      print("FAIL: Data has suspiciously few rows")
+    } else {
+      print("FAIL: Data has suspiciously many rows")
+    }
+  }
+
+#Checks if columns match the expected type
+
+test_stations_metadata_coltypes <-
+  function(df) {
+    expected_coltypes <-
+      c("character", "character", "double", "double", "double")
+    
+    if (all(df %>%
+            map_chr( ~ typeof(.)) == expected_coltypes) == TRUE) {
+      print("PASS: All cols have the correct specifications")
+    } else{
+      print("FAIL: Columns do not have the correct specification")
+    }
+  }
+  
+#Checks if the number of maximum missing values (200) are within the threshold
+
+test_stations_metadata_nmissing <-
+  function(df) {
+    max_miss_vals <- 200
+    
+    if (df %>% map_int( ~ sum(is.na((.)))) %>% sum(.) < max_miss_vals) {
+      print("PASS: Amount of missing values is reasonable")
+    } else {
+      print("FAIL: Too many missing values in data set")
+    }
+  }
+
+#Checks if correct timezone has been applied
+
+test_stations_metadata_latestdata_timezone <-
+  function(df) {
+    
+    if (attr(df$latestData,"tzone")=="UTC") {
+      print("PASS: latestData has UTC-time zone")
+    } else {
+      print("FAIL: latestData does not have expected UTC-time zone")
+    }
+  }
+
+
+#Calls the functions above 
+
+test_stations_metadata <- 
+  function(df){
+    test_stations_metadata_colnames(df)
+    test_stations_metadata_coltypes(df)
+    test_stations_metadata_nmissing(df)
+    test_stations_metadata_nrows(df)
+    test_stations_metadata_latestdata_timezone(df)
+  }
+

```

Comment:
You should have commented the existing file, not create a new one. The code still works fine, but it is unnecessary with duplicate files :)